mpy-cross/mpy_cross: Add a way to query the mpy version.

jimmo opened the pull request:
This returns the:
 - mpy_version
 - mpy_sub_version
 - native_arch
for files compiled with this mpy-cross binary.

I considered doing this by adding a different `--version` mode to mpy-cross and parsing that. Happy to do it either way (and open to suggestions!)

Also fixes some other issues in the mpy_cross Python wrapper, mostly the default path (all current uses specify the path explicitly though so that's why this wasn't a problem).

_This work was funded through GitHub Sponsors._

jimmo committed:
mpy-cross/mpy_cross: Fix default path to mpy-cross binary.

Needed to be updated to use build/mpy-cross.

Also fixes some other issues in the Python wrapper:
 - Rename find_mpy_cross_binary to _find_mpy_cross_binary
 - Fix passing of -march arg.
 - Decode stdout from subprocess.
 - Print stdout from mpy-cross in __main__.py.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>

jimmo committed:
mpy-cross/mpy_cross: Add a way to query the mpy version.

This returns the mpy version and sub-version for files compiled with this
mpy-cross binary.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>

dlech commented on mpy-cross/mpy_cross/__init__.py:
```diff
@@ -54,17 +55,40 @@
 
 
-def find_mpy_cross_binary(mpy_cross):
+def version(march=None, mpy_cross=None):
```
Might be better to call this `abi_version` to not confuse it with the mpy-cross/MicroPython version.

dlech replied:
Also, a doc string explaining the parameters and return value would be helpful.

dlech reviewed (commented):
What is the use case for including the arch in the version?

jimmo commented:
> What is the use case for including the arch in the version?

It's essentially a (complicated) way to map possible values of the `-march` flag to the resulting feature byte value.

jimmo committed:
mpy-cross/mpy_cross: Add list of architectures to `__all__`.

Signed-off-by: Jim Mussared <jim.mussared@gmail.com>

jimmo committed:
mpy-cross/mpy_cross: Add docstrings to public methods.

jimmo commented:
Thanks @dlech 

Updated to rename the new function to `mpy_version()` (we use "ABI" informally, but most things refer to this as the "mpy version"), and to parse the output of `--version` instead.

Also added a fix for the native arch constants (adding them to `__all__`), and docstrings.
